components/neeva/common/content_filtering_service.mojom

```diff
@@ -0,0 +1,46 @@
+// Copyright Neeva. All rights reserved.
+
+module neeva.mojom;
+
+enum ContentFilterMode {
+  BLOCK_COOKIES,
+  BLOCK_REQUESTS
+};
+
+struct ContentFilterRules {
+  ContentFilterMode mode;
+  array<string> top_level_host_exclusions;
+
+  // Where the rules can be found.
+  handle<platform> rules_data_fd;
+  uint64 rules_data_offset;
+  uint64 rules_data_size;
+};
+
+struct ContentFilterAction {
+  string host;
+};
+
+interface ContentFilterRulesProvider {
+  // Fetching rules:
+  // - On startup, child processes call RefreshRules passing a `current_generation_num`
+  //   of `0`, indicating that it has no rules currently.
+  // - Upon receiving new rules, child processes will immediately call `RefreshRules`
+  //   again this time passing the value of `new_generation_num` that they received.
+  //   The service will respond only when it has new rules to provide.
+  RefreshRules(int64 current_generation_num) =>
+      (int64 new_generation_num, ContentFilterRules rules);
```

Comment:
Might want to support letting `rules` be nullable to indicate that filtering is disabled.

Reply:
I decided to stick with the current approach. It keeps things simple.